library/modules/Items.cpp

```diff
@@ -590,6 +590,47 @@ df::item *Items::getContainer(df::item * item)
     return ref ? ref->getItem() : NULL;
 }
 
+void Items::getOuterContainerRef(df::specific_ref &spec_ref, df::item *item, bool init_ref)
+{
+    CHECK_NULL_POINTER(item);
+    //Reverse-engineered from ambushing unit code
+
+    if (init_ref)
+    {
+        spec_ref.type = specific_ref_type::ITEM_GENERAL;
+        spec_ref.data.object = item;
+    }
+
+    if (item->flags.bits.removed || !item->flags.bits.in_inventory)
+        return;
+
+    for (size_t i = 0; i < item->general_refs.size(); i++)
+    {
+        auto g = item->general_refs[i];
+        switch (g->getType())
+        {
+            case general_ref_type::CONTAINED_IN_ITEM:
+                if (auto item2 = g->getItem())
+                    return Items::getOuterContainerRef(spec_ref, item2);
+                break;
+            case general_ref_type::UNIT_HOLDER:
+                if (auto unit = g->getUnit())
+                    return Units::getOuterContainerRef(spec_ref, unit);
+                break;
+            default:
+                break;
+        }
+    }
+
+    auto s = findRef(item->specific_refs, specific_ref_type::VERMIN_ESCAPED_PET);
+    if (s)
+    {
+        spec_ref.type = specific_ref_type::VERMIN_EVENT;
+        spec_ref.data.vermin = s->data.vermin;
```

Comment:
Should be ``spec_ref.data.object = s->data.vermin;``?

Reply:
`spec_ref.data.vermin` should be fine, since `spec_ref.data` is a union.